File: frontend/src/app/core/services/ubicacion.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { CiudadResponse, PaisResponse } from '../models/ubicacion.model';

@Injectable({ providedIn: 'root' })
export class UbicacionService {
  constructor(private api: ApiService) {}

  listarPaises(): Observable<PaisResponse[]> {
    return this.api.get<PaisResponse[]>('/ubicaciones/paises');
  }

  listarCiudades(idPais: number): Observable<CiudadResponse[]> {
    return this.api.get<CiudadResponse[]>(`/ubicaciones/paises/${idPais}/ciudades`);
  }
}

File: backend/src/main/java/org/certificaciones/funproeibbackend/service/UbicacionService.java
package org.certificaciones.funproeibbackend.service;

import org.certificaciones.funproeibbackend.dto.CiudadResponse;
import org.certificaciones.funproeibbackend.dto.PaisResponse;

import java.util.List;

public interface UbicacionService {
    List<PaisResponse> listarPaises();
    List<CiudadResponse> listarCiudadesPorPais(Long idPais);
}
